### DISCUSSION TITLE
Having issues setting up tardis package as developer !

### DISCUSSION THREAD

**Discussion by Mr-MVP** [tardis-sn/tardis#2507] [Q&A] [answered]
I am getting messages that licenses has been deprecated and I should not use setup.py to setup the package. As by the pictures above. 

Even after this I imported tardis package and got this error message. 
![Screenshot from 2024-01-24 17-45-54](https://github.com/tardis-sn/tardis/assets/143615627/bae3dc7c-3de1-4014-a8df-12cd591915c0)
![Screenshot from 2024-01-24 17-46-16](https://github.com/tardis-sn/tardis/assets/143615627/d699b17f-0bf0-482e-8273-170342ee2b91)

I have read documentations lot of times but I wan't able to find an alternative to setup this package as developer. Can anybody help me with this issue ? It would be really appreciated

![Screenshot from 2024-01-24 17-49-07](https://github.com/tardis-sn/tardis/assets/143615627/37b51c9f-0390-4911-810d-fca845211b8f)

**Comment by andrewfullard** [comment]
Although the current install method is deprecated, it should function provided you successfully installed the environment from the lock file.

**Comment by Mr-MVP** [comment]
I successfully installed it from the lock file, conda-linux-64.lock using conda and created the environment using it. Then cloned my fork of the tardis repo via SSH then added remote upstream and fetched it, checked out upstream/master branch and then run

`python setup.py develop`

[All the steps are done according to the installation guide. ](https://tardis-sn.github.io/tardis/installation.html)

**Comment by andrewfullard** [comment, marked as answer]
There should not be a circular import. Did you actually load your environment before trying to import tardis? It looks like you are not running the python version from the tardis environment, instead using `/usr/lib` which is your system python.

**Reply by Mr-MVP** [reply]
I just wrote a basic code trial.py which imports tardis and run it through terminal 

![Screenshot from 2024-01-24 22-54-10](https://github.com/tardis-sn/tardis/assets/143615627/601ef67a-609f-492d-a802-674f6081773f)

I'l delete and try to setup the environment again. You're right about the environment issue, I'll look into it. Thanks for helping!
